cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -1422,6 +1424,41 @@ toTentativeEpochInfo (EraHistory _ interpreter) =
     $ hoistEpochInfo (first (Text.pack . show) . runExcept)
     $ Consensus.interpreterToEpochInfo (Consensus.unsafeExtendSafeZone interpreter)
 
+
+-- | Get slot number for timestamp, or an error if the UTC timestamp is before 'SystemStart' or after N+1 era
```

Comment:
N being the current era right?

Reply:
Yes, this function should work for current and next eras.